app/controllers/application_controller.rb

```diff
@@ -170,14 +170,21 @@ def update_cart # rubocop:disable MethodLength, AbcSize
     notices += "\n" unless notices.blank?
 
     # validate
-    errors = cart.validate_all
+    @errors = cart.validate_all
     # don't over-write flash if invalid date was set above
-    flash[:error] ||= notices + "\n" + errors.join("\n")
+    flash[:error] ||= notices + "\n" + @errors.join("\n")
     flash[:notice] = 'Cart updated.'
 
     # reload appropriate divs / exit
     prepare_catalog_index_vars if params[:controller] == 'catalog'
 
+    # # skip renders when called by submit_cart_updates_form
+    # return if params[:form]
+  end
+
+  # runs update cart and then reloads the javascript for the dates in sidebar
+  def reload_catalog_cart
```

Comment:
@AlanLiu96 Why did you split this function into two? when are they preformed separately?